DotNetDBF/MemoValue.cs

```diff
@@ -1,167 +1,156 @@
-
-                var raf =
-                    File.Open(aBase.DataMemoLoc,
-                              FileMode.OpenOrCreate,
-                              FileAccess.ReadWrite);
-
-                /* before proceeding check whether the passed in File object
-                    is an empty/non-existent file or not.
-                    */
-             
-
-                using (var tWriter = new BinaryWriter(raf, aBase.CharEncoding))
-                {
-                    if (raf.Length == 0)
-                    {
-                        var tHeader = new DBTHeader();
-                        tHeader.Write(tWriter);
-                    }
-
-                    var tValue = _value;
-                    if ((tValue.Length + sizeof(int)) % aBase.BlockSize != 0)
-                    {
-                        tValue = tValue + MemoTerminator;
-                    }
-
-                    var tPosition = raf.Seek(0, SeekOrigin.End); //Got To End Of File
-                    var tBlockDiff = tPosition%aBase.BlockSize;
-                    if (tBlockDiff != 0)
-                    {
-                        tPosition = raf.Seek(aBase.BlockSize - tBlockDiff, SeekOrigin.Current);
-                    }
-                    _block = tPosition/aBase.BlockSize;
-                    var tData = aBase.CharEncoding.GetBytes(tValue);
-                    var tDataLength = tData.Length;
-                    var tNewDiff = (tDataLength%aBase.BlockSize);
-                    tWriter.Write(tData);
-                    if (tNewDiff != 0)
-                        tWriter.Seek(aBase.BlockSize - (tDataLength % aBase.BlockSize), SeekOrigin.Current);
-                }
-            }
+            get { return _block; }
         }
 
-
         public string Value
         {
             get
             {
                 lock (_lockName)
                 {
-                 
                     if (!_new && !_loaded)
                     {
-                        using (var reader =new BinaryReader(
-                            File.Open(_fileLoc, 
-                            FileMode.Open, 
-                            FileAccess.Read,
-                            FileShare.Read)))
+                        using (BinaryReader reader = new BinaryReader(File.Open(_fileLoc, FileMode.Open, FileAccess.Read, FileShare.Read)))
                         {
-                            reader.BaseStream.Seek(_block*_base.BlockSize, SeekOrigin.Begin);
-                            string tString;
-                            var tStringBuilder = new StringBuilder();
-                            int tIndex;
-                            var tSoftReturn = _base.CharEncoding.GetString(new byte[] {0x8d, 0x0a});
+                            reader.BaseStream.Seek(_block * _base.BlockSize, SeekOrigin.Begin);
+                            string tempString;
+                            StringBuilder sb = new StringBuilder();

```

Comment:
C# coding conventions always use implicit typing (aka `var`)  when the type of the variable is obvious.